bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -3747,57 +3745,86 @@ void ReplicaImp::executeRequestsInPrePrepareMsg(concordUtils::SpanWrapper &paren
       LOG_DEBUG(CNSUS, "Consensus reached");
     }
     SCOPED_MDC("pp_msg_cid", ppMsg->getCid());
+    std::deque<IRequestsHandler::ExecutionRequest> accumulatedRequests;
     while (reqIter.getAndGoToNext(requestBody)) {
       size_t tmp = reqIdx;
       reqIdx++;
       if (!requestSet.get(tmp)) {
         continue;
       }
-
       ClientRequestMsg req((ClientRequestMsgHeader *)requestBody);
       if (req.flags() & EMPTY_CLIENT_FLAG) {
         continue;
       }
       SCOPED_MDC_CID(req.getCid());
       NodeIdType clientId = req.clientProxyId();
-
       uint32_t actualReplyLength = 0;
       uint32_t actualReplicaSpecificInfoLength = 0;
-      int status;
+      int preExecutedResponse = 1;
+      accumulatedRequests.push_back(IRequestsHandler::ExecutionRequest{
+          clientId,
+          static_cast<uint64_t>(lastExecutedSeqNum + 1),
+          req.flags(),
+          std::string(req.requestBuf(), req.requestLength()),
+          std::string(ReplicaConfig::instance().getmaxReplyMessageSize() - sizeof(ClientReplyMsgHeader), 0),
+          actualReplyLength,
+          actualReplicaSpecificInfoLength,
+          preExecutedResponse,
+          req.requestSeqNum()});
+
+      if (!ReplicaConfig::instance().blockAccumulation) {
```

Comment:
There is a lot of code duplication here... 
The only difference between block accumulation enabled/disabled should be in the way you call the new execute() method:
* blockAccumulation enabled: iterate over accumulatedRequests and call execute() for each of them separately
* blockAccumulation disabled: call execute() once with all the requests

Reply:
PS: see section 4.2 of the design document:
https://confluence.eng.vmware.com/display/BLOC/Block+accumulation+in+post-execution#Blockaccumulationinpost-execution-Post-executionoftheconsensusbatch

Reply:
At first, I accumulate all the requests in the preprepare message, then if the block accumulation feature is turned off I just sending a sublist of 1 request each time to the execute method.